### PR TITLE
Swift4.2

### LINKED ISSUE #163: Swift4.2 conversion
# What problem does this feture solve?
Swift 4.2 is now available.

# Describe what you've considered?
Let's convert some code to Swift4.2!

# Code sample / Spec
None.

### PR DESCRIPTION
**Pull request by usatie**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.

### Description of the Change
Converted from Swift 4.1.2 to 4.2
- Most of Range() is not needed any more
- Some name change by default

### Alternate Designs
Do nothing.

### Benefits
Modern language.

### Possible Drawbacks
Users who are using Xcode 9.0 or older cannot use BitcoinKit.
I'm not sure about this.

### Applicable Issues
closes #163

### CODE REVIEW

File: Sources/BitcoinKitPrivate/BitcoinKit.Private.swift [hunk 1]
```diff
@@ -225,23 +223,25 @@ public class _HDKey {
...
-        } else {
-            return nil
         }
+        
+        let fingerprintData = _Hash.sha256ripemd160(publicKey)
+        let fingerprint = fingerprintData.withUnsafeBytes{ (p: UnsafePointer<UInt32>) in
+            p.pointee
+        }
+        return _HDKey(privateKey: result,
+                      publicKey: result,
```
Comment (akifuj) [inline Sources/BitcoinKitPrivate/BitcoinKit.Private.swift]: Are really both of `privateKey` and `publicKey` the same value?

File: Tests/BitcoinKitTests/MockHelperTests.swift [hunk 2]
```diff
@@ -104,6 +104,12 @@ class MockHelperTests: XCTestCase {
             }
         }
     }
+    
+    func testP2SHRecursively() {
```
Comment (akifuj) [inline Tests/BitcoinKitTests/MockHelperTests.swift]: I don't think this test is necessary. If it is, please comment for the reason.

File: Tests/BitcoinKitTests/SerializationTests.swift [hunk 3]
```diff
@@ -0,0 +1,24 @@
+//
+//  SerializationTests.swift
```
Comment (akifuj) [inline Tests/BitcoinKitTests/SerializationTests.swift]: Please modify copy right.

File: Sources/BitcoinKitPrivate/BitcoinKit.Private.swift (hunk 1 above)
Comment (usatie) [reply Sources/BitcoinKitPrivate/BitcoinKit.Private.swift]: Oh this is kind of bad architecture of _HDKey.
I think _HDKey should be refactored, but it's not in the scope of this PR.

**Comment by DevilDimon** [comment]
Is there anything else that prevents this PR from being merged?

**Comment by usatie** [comment]
@DevilDimon Sorry for long time absence. Will merge this branch.

File: Sources/BitcoinKitPrivate/BitcoinKit.Private.swift [hunk 4]
```diff
@@ -140,11 +143,11 @@ public class _HDKey {
             data.append(0) // padding
             data += privateKey ?? Data()
```
Comment (usatie) [inline Sources/BitcoinKitPrivate/BitcoinKit.Private.swift]: If privateKey is nil, it should return nil

**Pull request merged**